src/MockHttpClient/HttpRequestMessageExtensions.cs

```diff
@@ -130,7 +132,104 @@ public static bool HasHeader(this HttpRequestMessage request, string headerName,
 
             return false;
         }
-    }
 
+        /// <summary>
+        /// Determines whether the request contains the specified content as string.
+        /// </summary>
+        /// <param name="request">the request</param>
+        /// <param name="expectedContent">the expected content</param>
+        /// <returns></returns>
+        public static bool HasStringContent(this HttpRequestMessage request, string expectedContent)
+        {
+            if (request.Content == null)
+            {
+                throw new ArgumentException($"string content '{expectedContent}' was expected but content is null");
+            }
 
+            var content = request.Content.ReadAsStringAsync().Result;
+
+            return content.Equals(expectedContent);
+        }
+
+        /// <summary>
+        /// Determines whether the request contains the specified content as json object.
+        /// </summary>
+        /// <typeparam name="T">The type of the expected content</typeparam>
+        /// <param name="request">the request</param>
+        /// <param name="expectedContent">the expected content</param>
+        /// <returns></returns>
+        public static bool HasJsonContent<T>(this HttpRequestMessage request, T expectedContent)
+        {
+            if (request.Content == null)
+            {
+                throw new ArgumentException($"json content '{expectedContent}' was expected but content is null");
+            }
+
+            var stringContent = request.Content.ReadAsStringAsync().Result;
+            var content = JsonConvert.DeserializeObject<T>(stringContent);
+
+            if (content == null)
+            {
+                throw new ArgumentException($"could not deserialize content to type {typeof(T)}");
```

Comment:
This would be better to return false than throwing an error. The condition will be checked for all requests going through the mock, and not all of them may deserialize correctly

Reply:
👍